src/main/java/net/fabricmc/accesswidener/RemappingDecorator.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2020 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.accesswidener;
+
+import org.objectweb.asm.commons.Remapper;
+
+/**
+ * Decorates a {@link net.fabricmc.accesswidener.AccessWidenerReader.Visitor} with a {@link Remapper}
+ * to remap names passing through the visitor if they come from a different namespace.
+ */
+public final class RemappingDecorator implements AccessWidenerReader.Visitor {
```

Comment:
AccessWidenerRemapper would be a better classname imo

Reply:
I'll rename it. Although I somewhat liked the theme of "chaining" decorators and filters together.

Reply:
Done